File: backend/app/schemas/emotion.py
from pydantic import BaseModel, Field
from typing import Dict, Optional

class EmotionRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000)
    language: Optional[str] = "auto"

class EmotionResponse(BaseModel):
    dominant_emotion: str
    dominant_confidence: float
    all_emotions: Dict[str, float]
    intensity: str  # low, medium, high
    recommendation: Optional[str] = None

class EmotionTimeline(BaseModel):
    timestamp: float
    emotions: Dict[str, float]

File: frontend/types/emotion.ts
export interface EmotionRequest {
  text: string
  language?: string
}

export interface EmotionResponse {
  dominant_emotion: string
  dominant_confidence: number
  all_emotions: Record<string, number>
  intensity: 'low' | 'medium' | 'high'
  recommendation?: string
}

export interface EmotionTimeline {
  timestamp: string
  emotion: string
  confidence: number
  intensity: string
}